libnetwork/agent.go

```diff
@@ -628,9 +628,15 @@ func (ep *Endpoint) addServiceInfoToCluster(sb *Sandbox) error {
 		return nil
 	}
 
-	name := ep.Name()
-	if ep.isAnonymous() {
-		name = ep.MyAliases()[0]
+	ep.mu.Lock()
+	dnsNames := make([]string, len(ep.dnsNames))
+	copy(dnsNames, ep.dnsNames)
+	ep.mu.Unlock()
+
+	primaryDNSName := dnsNames[0]
+	var dnsAliases []string
+	if len(dnsNames) > 1 {
+		dnsAliases = dnsNames[1:]
 	}
```

Comment:
```suggestion
	primaryDNSName, dnsAliases := dnsNames[0], dnsNames[1:]
```
for consistency with line 713